check-cla/action.yml

```diff
@@ -3,161 +3,196 @@ name: CLA check
-      uses: actions/github-script@v6
+    # if triggered by a comment, leave a reaction
+    - name: React to comment
+      uses: peter-evans/create-or-update-comment@v3.0.0
+      if: github.event_name == 'issue_comment'
       with:
-        script: |
-          const { owner, repo, number } = context.issue;
-          const pullRequest = await github.rest.pulls.get({
-            owner,
-            repo,
-            pull_number: number,
-          });
-          console.log(pullRequest);
-          const sha = pullRequest.data.head.sha;
-          console.log(sha);
-          core.setOutput('sha', sha);
-
-          const labels = pullRequest.data.labels.map(label => label.name)
-          console.log(labels);
-          core.setOutput('labels', labels);
-
-          const hasLabel = labels.includes('${{ inputs.label }}')
-          console.log(hasLabel);
-          core.setOutput('hasLabel', hasLabel);
+        token: ${{ inputs.token }}
+        comment-id: ${{ github.event.comment.id }}
+        reactions: eyes
 
+    # commit status → pending
     - name: Set commit status with pending
-      uses: dholth/github-status-action@runs-using-node16
+      uses: conda/actions/set-commit-status@customize-cla-repo
       with:
-        authToken: ${{ inputs.token }}
+        token: ${{ inputs.token }}
         context: CLA check
         description: Checking conda CLA...
         state: pending
-        sha: ${{ steps.pr.outputs.sha || github.sha }}
 
-    - name: Check if current actor has signed
+    # has_label, number, contributor, url, has_signed
+    - name: Collect PR metadata
       uses: actions/github-script@v6
-      id: contributors
+      id: metadata
       with:
         github-token: ${{ inputs.token }}
         script: |
-          console.log(context);
-          const getContributors = async () => {
-            try {
-              const results = (
-                await github.rest.repos.getContent({
-                  owner: 'conda',
-                  repo: 'infrastructure',
-                  path: '.clabot'
-                })
-              );
-              return JSON.parse(Buffer.from(results.data.content, results.data.encoding).toString('utf-8')).contributors;
-            } catch (err) {
-              core.error(`Could not retrieve contributors, returning undefined. Reason: ${err}`)
-              return undefined;
-            }
-          }
-          const contributors = await getContributors();
-          console.log(contributors);
-          const pull_request = (context.payload.issue || context.payload.pull_request || context.payload);
-          const creator = pull_request.user.login;
-          console.log(creator);
-          const hasSigned = contributors.includes(creator);
-          console.log(hasSigned);
-          core.setOutput('contributors', contributors);
-          core.setOutput('hasSigned', hasSigned);
-
-    # add [cla-signed] label if actor has already signed
-    - name: Add label
```

Comment:
Could we add this to a separate javascript file? This is started to get long enough to warrant that, imo.

Reply:
It's a common pattern for actions to include all JS inlined, even if it's not great, let's follow that for now